core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncResponseTransformer.java

```diff
@@ -44,13 +49,48 @@
     private volatile AsynchronousFileChannel fileChannel;
     private volatile CompletableFuture<Void> cf;
     private volatile ResponseT response;
+    private final long position;
+    private final FileWriteOption fileModificationOption;
+    private final boolean deleteOnFailure;
 
     public FileAsyncResponseTransformer(Path path) {
         this.path = path;
+        this.fileModificationOption = FileWriteOption.CREATE_NEW;
+        this.deleteOnFailure = true;
+        this.position = 0L;
+    }
+
+    public FileAsyncResponseTransformer(Path path, FileTransformerConfiguration fileConfiguration) {
+        this.path = path;
+        this.fileModificationOption = fileConfiguration.fileModificationOption().orElse(FileWriteOption.CREATE_NEW);
+        this.position = determineFilePositionToWrite(path);
+        this.deleteOnFailure = fileConfiguration.deleteOnFailure().orElse(true);
+    }
+
+    private long determineFilePositionToWrite(Path path) {
+        if (fileModificationOption == CREATE_OR_APPEND_EXISTING) {
+            try {
+                return Files.size(path);
+            } catch (NoSuchFileException e) {
+                // Ignore
+            } catch (IOException exception) {
+                throw SdkClientException.create("Cannot determine the current file size " + path, exception);
+            }
+        }
+        return  0L;
     }
 
     private AsynchronousFileChannel createChannel(Path path) throws IOException {
-        return AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
+        switch (fileModificationOption) {
+            case CREATE_OR_APPEND_EXISTING:
+                return AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE);
+            case CREATE_OR_REPLACE_EXISTING:
+                return AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE,
+                                                    StandardOpenOption.TRUNCATE_EXISTING);
+            case CREATE_NEW:
+            default:
+                return AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
```

Comment:
Imo, we might want to fail by default to make sure it's an explicit choice when we add future modification options.